src/sentry/static/sentry/app/views/settings/components/forms/inputField.tsx

```diff
@@ -14,14 +14,25 @@ type Props = {
     'value' | 'placeholder' | 'disabled' | 'onBlur'
   >;
 
+export type onEvent = (value: any, event?: React.FormEvent<HTMLInputElement>) => void;
```

Comment:
normally, we avoid having to export types, so we use React.ComponentProps<typeof x>... are you sure that you have to export this type?

Reply:
Can we maybe use a generic type for the value?

Reply:
From my (limited) understanding of using `ComponentProps` to retrieve types, we prefer using it to encapsulate the types used in a component as part of it's props to avoid having multiple exports for every component, right?

But `onEvent` isn't directly part of the component and making it part of the props just so you can export it seems backwards.

Reply:
yes but  you are using the onEvent in the onChange, onBlur... please see: 

https://github.com/getsentry/sentry/blob/8d1fba8a1300f9445bc1bed457014c9ffcf85052/src/sentry/static/sentry/app/views/settings/components/forms/inputField.tsx#L32

I thought that you could take the type from one of this props.. but I agree with you, I am also not a big fan of ComponentProps...